aineko/core/dataset.py

```diff
@@ -1,525 +1,276 @@
-
-        if how == "next":
-            remaining = len(self.values)
-            if remaining > 0:
-                if remaining == 1:
-                    self.empty = True
-                return {
-                    "timestamp": datetime.datetime.now().strftime(
-                        AINEKO_CONFIG.get("MSG_TIMESTAMP_FORMAT")
-                    ),
-                    "message": self.values.pop(0),
-                    "source_node": "test",
-                    "source_pipeline": "test",
-                }
-        if how == "last":
-            if self.values:
-                return self.values[-1]
-
-        return None
-
-    def next(self) -> Optional[dict]:
-        """Wraps `consume(how="next")`, blocks until available.
+        def _write(self, **kwargs) -> Any:
+            pass
 
-        Returns:
-            msg: message from the dataset
-        """
-        return self.consume(how="next")
+        def _create(self, **kwargs) -> Any:
+            pass
 
-    def last(self, timeout: float = 1) -> Optional[dict]:
-        """Wraps `consume(how="last")`, blocks until available.
+        def _delete(self, **kwargs) -> Any:
+            pass
 
-        Returns:
-            msg: message from the dataset
-        """
-        return self.consume(how="last", timeout=timeout)
+        def _initialize(self, **kwargs) -> Any:
+            pass
 
+        def _exists(self, **kwargs) -> bool:
+            pass
 
-class FakeDatasetProducer:
-    """Fake dataset producer for testing purposes.
+        def _describe(self, **kwargs) -> str:
+            pass
+    ```
 
-    The class will store in its state the list of values produced
-    by the node.
+    If `MyDataset` was defined in the file
+    `./aineko/datasets/mydataset.py`, a new dataset
+    can be created using the `from_config` method:
 
-    Args:
-        dataset_name: name of the dataset
-        node_name: name of the node that is producing the dataset
+    Example:
+    ```python
 
-    Attributes:
-        dataset_name (str): name of the dataset
-        node_name (str): name of the node that is producing the dataset
-        values (list): list of mock data values produced by the node
+    dataset = AbstractDataset.from_config(
+        name="my_dataset_instance",
+        config={
+            "type": "aineko.datasets.mydataset.MyDataset",
+            "location": "foo",
+            "params": {
+                "param_1": "bar"
+            }
+        }
+    )
+    ```
     """
 
-    def __init__(self, dataset_name: str, node_name: str):
-        """Initialize the producer."""
-        self.dataset_name = dataset_name
-        self.node_name = node_name
-        self.values = []  # type: ignore
+    name: str
 
-    def produce(self, message: Any) -> None:
-        """Stores message in self.values.
+    @classmethod
+    def from_config(cls: Type[A], name: str, config: Dict[str, Any]) -> A:
```

Comment:
```suggestion
    def from_config(cls, name: str, config: Dict[str, Any]) -> AbstractDataset:
```
cls doesn't need to be type hinted and we know that from_config will return AbstractDataset

Reply:
I get mypy errors when I try implementing it like this (remove Type[A] and change return type to AbstractDataset):
```Undefined variable 'AbstractDataset' (undefined-variable)```

If I try to remove Type[A] and replace the return type as A, I get a different mypy error:
```error: A function returning TypeVar should receive at least one argument containing the same TypeVar  [type-var]```

I think this has to do with from_config returning a subclass of AbstractDataset

Reply:
Oh I see, if I put quotes around AbstractDataset for the return type (and remove Type[A]), it doesn't throw mypy errors. 
I do think what you propose is simpler, but also might have slightly less type safety.